pkg/instancetype/controller/vm/controller_test.go

```diff
@@ -0,0 +1,1052 @@
+						Guest: instancetypeObj.Spec.CPU.Guest,
+					},
+					Memory: v1alpha1.MemoryInstancetype{
+						Guest: instancetypeObj.Spec.Memory.Guest,
+					},
+				}
+
+				specBytes, err := json.Marshal(&v1alpha1instancetypeSpec)
+				Expect(err).ToNot(HaveOccurred())
+
+				specRevision := v1alpha1.VirtualMachineInstancetypeSpecRevision{
+					APIVersion: "",
+					Spec:       specBytes,
+				}
+				specRevisionBytes, err := json.Marshal(specRevision)
+				Expect(err).ToNot(HaveOccurred())
+
+				return specRevisionBytes
+			}),
+			Entry("using v1alpha1", func() []byte {
+				v1alpha1instancetype := &v1alpha1.VirtualMachineInstancetype{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: v1alpha1.SchemeGroupVersion.String(),
+						Kind:       "VirtualMachineInstancetype",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: instancetypeObj.Name,
+					},
+					Spec: v1alpha1.VirtualMachineInstancetypeSpec{
+						CPU: v1alpha1.CPUInstancetype{
+							Guest: instancetypeObj.Spec.CPU.Guest,
+						},
+						Memory: v1alpha1.MemoryInstancetype{
+							Guest: instancetypeObj.Spec.Memory.Guest,
+						},
+					},
+				}
+				instancetypeBytes, err := json.Marshal(v1alpha1instancetype)
+				Expect(err).ToNot(HaveOccurred())
+
+				return instancetypeBytes
+			}),
+			Entry("using v1alpha2", func() []byte {
+				v1alpha2instancetype := &v1alpha2.VirtualMachineInstancetype{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: v1alpha2.SchemeGroupVersion.String(),
+						Kind:       "VirtualMachineInstancetype",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: instancetypeObj.Name,
+					},
+					Spec: v1alpha2.VirtualMachineInstancetypeSpec{
+						CPU: v1alpha2.CPUInstancetype{
+							Guest: instancetypeObj.Spec.CPU.Guest,
+						},
+						Memory: v1alpha2.MemoryInstancetype{
+							Guest: instancetypeObj.Spec.Memory.Guest,
+						},
+					},
+				}
+				instancetypeBytes, err := json.Marshal(v1alpha2instancetype)
+				Expect(err).ToNot(HaveOccurred())
+
+				return instancetypeBytes
+			}),
+			Entry("using v1beta1", func() []byte {
+				instancetypeBytes, err := json.Marshal(instancetypeObj)
+				Expect(err).ToNot(HaveOccurred())
+
+				return instancetypeBytes
+			}),
+		)
+
+		It("should sync correctly if an existing ControllerRevision is present but not referenced by InstancetypeMatcher", func() {
+			instancetypeRevision, err := instancetype.CreateControllerRevision(vm, instancetypeObj)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = virtClient.AppsV1().ControllerRevisions(vm.Namespace).Create(
+				context.Background(), instancetypeRevision, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vm.Spec.Instancetype = &virtv1.InstancetypeMatcher{
+				Name: instancetypeObj.Name,
+				Kind: instancetypeapi.SingularResourceName,
+			}
+
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Create(context.TODO(), vm, metav1.CreateOptions{})
+			Expect(err).To(Succeed())
+
+			sanitySync(vm, vmi)
```

Comment:
Maybe call the func `sanityExecute` to be consistent with other parts of the code base?

Reply:
ACK, a misunderstanding on my part then.